### ISSUE TITLE
[BUG] New Access Control feature returns error on Default rule

### ISSUE THREAD

**Issue by Mereck13579** [tobychui/zoraxy#130]
**Describe the bug**
I have 2 access control rules in addition to the default one. The 2 additional are working perfectly fine, but using the default one returns me an error on both Firefox and Chrome.
Chrome gets a "This site can’t be reached" ERR_HTTP2_PROTOCOL_ERROR, Firefox gets a "Secure Connection Failed" NS_ERROR_NET_INTERRUPT.
For testing purposes I made both default and another rule exactly the same, default rules return error while the other do not.
If I deactivate the rules, I get properly redirected to the Zoraxy static page without any SSL error.

**To Reproduce**
1. Set up a default rule with some basic Whitelist (IP ranges)
2. Set up a new rule with the exact same Whitelist
3. Go to HTTP Proxy, set a host on the default access list
5. Set the host with the new rule
6. Proxy is working properly
7. Deactivate the host proxy rule
8. See Zoraxy default static page

**Expected behavior**
Web page returned

**Screenshots**
![Screenshot 2024-04-26 at 15-07-36 Control Panel Zoraxy](https://github.com/tobychui/zoraxy/assets/7541087/a89400f3-754e-4885-afb0-bc3683c361bd)
![Screenshot 2024-04-26 at 15-07-25 Control Panel Zoraxy](https://github.com/tobychui/zoraxy/assets/7541087/c35eec50-6ae7-447b-9069-c17b234b548e)

**Browser (if it is a bug appears on the UI section of the system):**
 - OS: Windows 11 / Android 14
 - Browser / Version Chrome 123.0.6312.86 and Firefox 115.10.0esr

**Host Environment (please complete the following information):**
-  Arch: x86_64
 - OS: Alpine Linux
 - Version 3.19
 - Docker version 26.0.0

**Additional context**
To check the error you can try my testing FQDN: https://test.yangi.re
Error happens on both internal and external networks (but both belonging to the whitelisted IPs)

**Comment by Mereck13579** [comment]
Found the following workaround: deactivating the white list seems to make it work

**Comment by tobychui** [comment]
Hi @Mereck13579 , thanks for the issue report. I will try to figure out what is the issue this weekend.

**Comment by Mereck13579** [comment]
Unsure if it's related or not but it seems like the whitelist rules are not working properly as well at times... 

I have a rule to limit some proxies to local addresses only (192.168.20.0/24 in the whitelist, my PC IP is 192.168.20.42) but I get a 403 forbidden when trying to access it. Disabling the white list makes it work.

Tell me if you want me to open another ticket instead for this.

**Comment by tobychui** [comment]
@Mereck13579 Interesting, as I cannot reproduce this in my Zoraxy test environment, can you help me try out something?

Edit `src/mod/access/whitelist.go` and recompile zoraxy using `go mod tidy` and `go build`

```go
//Line 3
import (
	"fmt" //Add this line to import fmt
	"strings"

	"imuslab.com/zoraxy/mod/netutils"
)
```

```go
func (s *AccessRule) IsIPWhitelisted(ipAddr string) bool {
	//Check for IP wildcard and CIRD rules
	WhitelistedIP := *s.WhiteListIP
	for ipOrCIDR, _ := range WhitelistedIP {
		wildcardMatch := netutils.MatchIpWildcard(ipAddr, ipOrCIDR)
		if wildcardMatch {
			return true
		}

		cidrMatch := netutils.MatchIpCIDR(ipAddr, ipOrCIDR)
		fmt.Println(ipAddr, cidrMatch) //Add this line after line 91
		if cidrMatch {
			return true
		}
	}

	return false
}
```

Then we will know if your rule is set correctly or there are issue with the CIDR matching logic (which is rare as it is golang's build in function)

**Comment by Mereck13579** [comment]
I modified the files accordingly in my Zoraxy container image and recreated the image but I'm unsure on where I should execute the go mod tidy and go build commands, is that inside of the container itself afterwards?

**Comment by tobychui** [comment]
@Mereck13579 Oh uh you running Zoraxy in container. In that case, you will need a go compiler inside the container or you build it outside of your container and use SFTP to replace the running binary file in the container.

**Comment by tobychui** [comment]
Fixed in v3.0.3

**Comment by Mereck13579** [comment]
Just to add up on the closure, since I didn't feel confident enough to build outside and push it to container, I just set up a new VM for Zoraxy for testing purposes. I could reproduce the issue with the latest 3.0.3 version, but as I was trying to diagnose it further (with tcpdump) I realized that despite having my local DNS returning the local IP address for the test FQDN the traffic incoming on the VM actually came from OUTSIDE the LAN, so my IPs were effectively not the one I was expecting on the Zoraxy instance... I am currently investingating why that's the case but it means it is outside of Zoraxy's scope.

**Comment by Mereck13579** [comment]
Wrapping it up as I finally found the culprit, since it might be useful to anyone encountering the same issue.
Basically the problem came from Chrome. By default I had "Secure" DNS enabled, effectively bypassing my own local DNS so instead of having test.yangi.re being redirected to 192.168.20.XXX, I had Chrome ask a public DNS about it, returning my public IP instead. This resulted in my request going through the local router, going outside (to my external router) and going back in, effectively coming back with my external router IP instead of mine.
tl;dr, if you use Chrome, go to chrome://settings, search for "DNS" and check if this thing is disabled:
![image](https://github.com/tobychui/zoraxy/assets/7541087/8b635749-fd37-4219-b913-049abef55d36)
